dashboard/modules/job/tests/test_sdk.py

```diff
@@ -142,5 +154,88 @@ def test_temporary_uri_reference(monkeypatch, expiration_s):
                 print("Internal KV was GC'ed at time ", time.time() - start)
 
 
+@pytest.fixture
+def mock_candidate_number():
+    os.environ["CANDIDATE_AGENT_NUMBER"] = "2"
+    yield
+    os.environ.pop("CANDIDATE_AGENT_NUMBER", None)
+
+
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [{"include_dashboard": True}], indirect=True
+)
+def test_job_head_choose_job_agent_E2E(mock_candidate_number, ray_start_cluster_head):
+    cluster = ray_start_cluster_head
+    assert wait_until_server_available(cluster.webui_url) is True
+    webui_url = cluster.webui_url
+    webui_url = format_web_url(webui_url)
+    client = JobSubmissionClient(webui_url)
+
+    def get_register_agents_number():
+        response = requests.get(webui_url + "/internal/node_module")
+        response.raise_for_status()
+        result = response.json()
+        data = result["data"]
+        return data["registeredAgents"]
+
+    def submit_job_and_wait_finish():
+        submission_id = client.submit_job(entrypoint="echo hello")
+
+        wait_for_condition(_check_job_succeeded, client=client, job_id=submission_id)
+
+    # make sure list(cluster.worker_nodes)[0] will be the owner of a supervisor actor.
```

Comment:
Just a clarification question, how do we ensure the supervisor actor doesn't get scheduled on the head node?